File: LeetCode/Kai/82RemoveDuplicatesFromSortedListII.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        
        ListNode p = dummy;
        while(p.next!=null && p.next.next!=null) {
            if(p.next.val==p.next.next.val) {
                int duplicate = p.next.val;
                while(p.next!=null && p.next.val==duplicate) p.next = p.next.next;
            } else {
                p = p.next;
            }
        }
        return dummy.next;
    }
    
}

File: LeetCode/Peng/82RemoveDuplicatesFromSortedListII.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (!head || !head->next) return head;
        ListNode* dummy = new ListNode(-1);
        ListNode* current = dummy;
        while (head && head->next) {
            if (head->val == head->next->val) {
                int temp = head->val;
                while (head && head->val == temp) {
                    head = head->next;
                }
            } else { 
                current->next = head;
                current = current->next;
                head = head->next;
            }
        }
        current->next = head;
        return dummy->next;
    }
};